src/main.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Main module to run the prometheus exporter server."""
+
+import argparse
+import logging
+import os
+import sys
+from wsgiref.simple_server import make_server
+
+from prometheus_client import make_wsgi_app
+from prometheus_client.core import REGISTRY
+
+from src.collector import Config, DashBoardsCollector
+
+
+def setup_logging() -> None:
+    """Setup the log for the application."""
+    logging.basicConfig(
+        level=logging.INFO,
+        format="%(asctime)s - %(levelname)s - %(message)s",
+        handlers=[logging.StreamHandler()],
+    )
+
+
+def parse_command_line(args: list[str]) -> argparse.Namespace:
+    """Command line parser.
+
+    Args:
+        args (list[str]): List of arguments to parse
+
+    Returns:
+        argparse.Namespace: Command line arguments.
+    """
+    parser = argparse.ArgumentParser(
+        prog=__package__,
+        description=__doc__,
+    )
+    parser.add_argument(
+        "url",
+        type=str,
+        help="The OpenSearch address to fetch metrics from. E.g: http(s)://<IP>:5601",
+    )
```

Comment:
I think it makes more sense to also make this one optional and default to `http://localhost:5601`. Exporters are usually living in the same env as the workload.